js/src/edit-block/components/front-end-preview.js

```diff
@@ -0,0 +1,72 @@
+/**
+ * External dependencies
+ */
+import * as React from 'react';
+
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { PostSavedState } from '@wordpress/editor';
+import { __ } from '@wordpress/i18n';
+import ServerSideRender from '@wordpress/server-side-render';
+
+/**
+ * Internal dependencies
+ */
+import { getBlock, getBlockNameWithNameSpace } from '../helpers';
+
+/**
+ * The front-end preview of the block.
+ *
+ * Gets the saved post with .getCurrentPost(),
+ * not the edited post.
+ * The server side only has access to the saved post.
+ * So there can be an error in <ServerSideRender>
+ * if it passes attributes that aren't yet saved.
+ *
+ * @return {React.ReactElement} The front-end preview.
+ */
+const FrontEndPreview = () => {
```

Comment:
This file is mainly extracted from `editor.js`, with some conditionals in case there's nothing to preview.